casaos-healthcheck/run.sh

```
@@ -140,23 +140,152 @@ check_dns_resolution() {
+        print_color "0;31" "${CROSS_MARK} Memory usage is high: ${used_mem}MB used of ${total_mem}MB (Threshold: ${threshold}MB)"
+    else
+        print_color "0;32" "${CHECK_MARK} Memory usage is acceptable: ${used_mem}MB used of ${total_mem}MB (Threshold: ${threshold}MB)"
+    fi
+}
+
+# Function to check system temperature
+check_system_temperature() {
+    print_header "System Temperature Check"
+    if command -v sensors &> /dev/null; then
+        sensors | grep -E 'Core|temp' | while read -r line; do
+            local temp=$(echo "$line" | awk '{print $2}' | sed 's/+//' | sed 's/°C//')
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)